python-package/xgboost/spark/model.py

```diff
@@ -1,28 +1,73 @@
-# type: ignore
 """Xgboost pyspark integration submodule for model API."""
 # pylint: disable=fixme, invalid-name, protected-access, too-few-public-methods
 import base64
+import logging
 import os
 import uuid
-
-from pyspark import SparkFiles, cloudpickle
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
+
+from pyspark import SparkContext, SparkFiles, cloudpickle
 from pyspark.ml.util import DefaultParamsReader, DefaultParamsWriter, MLReader, MLWriter
 from pyspark.sql import SparkSession
 
 from xgboost.core import Booster
 
 from .utils import get_class_name, get_logger
 
-
-def _get_or_create_tmp_dir():
+if TYPE_CHECKING:
+    from .. import XGBClassifier, XGBRanker, XGBRegressor
+    from . import (
+        SparkXGBClassifier,
+        SparkXGBClassifierModel,
+        SparkXGBRanker,
+        SparkXGBRankerModel,
+        SparkXGBRegressor,
+        SparkXGBRegressorModel,
+    )
+
+    # xgboost types
+    XGB_ESTIMATOR = Union[XGBClassifier, XGBRanker, XGBRegressor]
```

Comment:
Use `XGBModel` class instead. It's the parent class for all referenced classes here.